### PR TITLE
[DTSSTCI-117] Audit Unit tests and update for Caseworker, CicCase and Citizen Modules

### PR DESCRIPTION
**Pull request by oleon23**
Review and update/improvement of unit tests for Caseworker, CicCase and Citizen modules

Ticket:
https://tools.hmcts.net/jira/browse/DTSSTCI-117

- [x] tests have been updated / new tests has been added (if needed)

### CODE REVIEW

**Review by Jefex7 — COMMENTED** [review]
Rajaram mentionned on my pr that we might want to leave the comments (Given,When,Then)  for now as a lot of test have them for better readability and if needed we can remove them later.

File: src/main/java/uk/gov/hmcts/sptribs/caseworker/event/page/HearingRecordingUploadPage.java [hunk 1]
```diff
@@ -59,7 +59,7 @@ public AboutToStartOrSubmitResponse<CaseData, State> midEvent(CaseDetails<CaseDa
                     if (StringUtils.isEmpty(documentListValue.getValue().getDocumentEmailContent())) {
                         errors.add("Description is mandatory for each document");
                     }
-                    if (null == documentListValue.getValue().getDocumentCategory()) {
+                    if (documentListValue.getValue().getDocumentCategory() == null) {
```
Comment (tomxelliott) [inline src/main/java/uk/gov/hmcts/sptribs/caseworker/event/page/HearingRecordingUploadPage.java:62]: could use isNull() method here

File: src/main/java/uk/gov/hmcts/sptribs/caseworker/event/page/HearingRecordingUploadPage.java [hunk 2]
```diff
@@ -48,9 +48,9 @@ public AboutToStartOrSubmitResponse<CaseData, State> midEvent(CaseDetails<CaseDa
         final CaseData data = details.getData();
         List<ListValue<CaseworkerCICDocument>> uploadedDocuments = data.getListing().getSummary().getRecFile();
         List<String> errors = new ArrayList<>();
-        if (null != uploadedDocuments) {
+        if (uploadedDocuments != null) {
             for (ListValue<CaseworkerCICDocument> documentListValue : uploadedDocuments) {
-                if (null == documentListValue.getValue().getDocumentLink()) {
+                if (documentListValue.getValue().getDocumentLink() == null) {
```
Comment (tomxelliott) [inline src/main/java/uk/gov/hmcts/sptribs/caseworker/event/page/HearingRecordingUploadPage.java:53]: could use isNull() method here

File: src/main/java/uk/gov/hmcts/sptribs/document/DocumentUtil.java [hunk 3]
```diff
@@ -94,15 +94,15 @@ public static List<String> validateUploadedDocuments(List<ListValue<CaseworkerCI
 
         if (uploadedDocuments != null && !uploadedDocuments.isEmpty()) {
             for (ListValue<CaseworkerCICDocument> documentListValue : uploadedDocuments) {
-                if (null == documentListValue.getValue().getDocumentLink()) {
+                if (documentListValue.getValue().getDocumentLink() == null) {
```
Comment (tomxelliott) [inline src/main/java/uk/gov/hmcts/sptribs/document/DocumentUtil.java:97]: could use isNull() method here

File: src/main/java/uk/gov/hmcts/sptribs/document/DocumentUtil.java [hunk 4]
```diff
@@ -94,15 +94,15 @@ public static List<String> validateUploadedDocuments(List<ListValue<CaseworkerCI
...
                     errors.add("Please attach the document");
                 } else {
                     errors.addAll(validateCaseworkerCICDocumentFormat(List.of(documentListValue)));
 
                     if (StringUtils.isEmpty(documentListValue.getValue().getDocumentEmailContent())) {
                         errors.add("Description is mandatory for each document");
                     }
-                    if (null == documentListValue.getValue().getDocumentCategory()) {
+                    if (documentListValue.getValue().getDocumentCategory() == null) {
```
Comment (tomxelliott) [inline src/main/java/uk/gov/hmcts/sptribs/document/DocumentUtil.java:105]: could use isNull() method here

File: src/test/java/uk/gov/hmcts/sptribs/caseworker/event/page/HearingRecordingUploadPageTest.java [hunk 5]
```diff
@@ -27,80 +28,95 @@ public class HearingRecordingUploadPageTest {
...
 
     @Test
-    void shouldValidateUploadedDocumentWithoutDoc() {
-        //Given
-        final CaseDetails<CaseData, State> caseDetails = new CaseDetails<>();
-        final Listing listing = getRecordListing();
-        List<ListValue<CaseworkerCICDocument>> documentList = new ArrayList<>();
-        final CaseworkerCICDocument document = CaseworkerCICDocument.builder()
+    void midEventReturnsErrorForMissingDocumentLink() {
+        this.documentList = new ArrayList<>();
```
Comment (tomxelliott) [inline src/test/java/uk/gov/hmcts/sptribs/caseworker/event/page/HearingRecordingUploadPageTest.java:101]: is this required? should be fine without

File: src/main/java/uk/gov/hmcts/sptribs/caseworker/event/page/HearingRecordingUploadPage.java (hunk 2 above)
Comment (oleon23) [reply src/main/java/uk/gov/hmcts/sptribs/caseworker/event/page/HearingRecordingUploadPage.java:53]: Do you mean use: `Objects.isNull()`?

File: src/test/java/uk/gov/hmcts/sptribs/caseworker/event/page/HearingRecordingUploadPageTest.java (hunk 5 above)
Comment (oleon23) [reply src/test/java/uk/gov/hmcts/sptribs/caseworker/event/page/HearingRecordingUploadPageTest.java:101]: It tests the if clause of `HearingRecordUploadPage.midEvent()`, line 55:
https://github.com/hmcts/sptribs-case-api/blob/1d9be6021eb4f88250f8179e54d6f1d609c92d99/src/main/java/uk/gov/hmcts/sptribs/caseworker/event/page/HearingRecordingUploadPage.java#L55

File: src/test/java/uk/gov/hmcts/sptribs/caseworker/event/page/IssueDecisionSelectRecipientsTest.java [hunk 6]
```diff
@@ -56,11 +58,28 @@ void shouldBeInvalidIfNoRecipientsSelected() {
...
 
-        //When
         final AboutToStartOrSubmitResponse<CaseData, State> response = selectRecipients.midEvent(caseDetails, caseDetails);
 
-        //Then
+        assertNull(response.getData().getCicCase().getNotifyPartySubject());
+        assertNull(response.getData().getCicCase().getNotifyPartyRepresentative());
+        assertNull(response.getData().getCicCase().getNotifyPartyRespondent());
+        assertNull(response.getData().getCicCase().getNotifyPartyApplicant());
         assertThat(response.getErrors()).hasSize(1);
```
Comment (gmmagruder) [inline src/test/java/uk/gov/hmcts/sptribs/caseworker/event/page/IssueDecisionSelectRecipientsTest.java:67]: Could add an assert for expected error message returned.

File: src/test/java/uk/gov/hmcts/sptribs/caseworker/event/page/CloseCaseSelectRecipientsTest.java [hunk 7]
```diff
@@ -22,36 +29,48 @@ public class CloseCaseSelectRecipientsTest {
...
 
-        //When
         final AboutToStartOrSubmitResponse<CaseData, State> response = selectRecipients.midEvent(caseDetails, caseDetails);
 
-        //Then
+        assertNull(response.getData().getCicCase().getNotifyPartySubject());
+        assertNull(response.getData().getCicCase().getNotifyPartyRepresentative());
+        assertNull(response.getData().getCicCase().getNotifyPartyRespondent());
+        assertNull(response.getData().getCicCase().getNotifyPartyApplicant());
         assertThat(response.getErrors()).hasSize(1);
```
Comment (gmmagruder) [inline src/test/java/uk/gov/hmcts/sptribs/caseworker/event/page/CloseCaseSelectRecipientsTest.java:59]: Could add an assert for expected error message returned.

File: src/test/java/uk/gov/hmcts/sptribs/caseworker/event/page/IssueFinalDecisionSelectRecipientsTest.java [hunk 8]
```diff
@@ -58,35 +59,47 @@ void shouldBeInvalidIfNoRecipientsSelected() {
...
 
-        //When
         final AboutToStartOrSubmitResponse<CaseData, State> response = selectRecipients.midEvent(caseDetails, caseDetails);
 
-        //Then
+        assertNull(response.getData().getCicCase().getNotifyPartySubject());
+        assertNull(response.getData().getCicCase().getNotifyPartyRepresentative());
+        assertNull(response.getData().getCicCase().getNotifyPartyRespondent());
+        assertNull(response.getData().getCicCase().getNotifyPartyApplicant());
         assertThat(response.getErrors()).hasSize(1);
```
Comment (gmmagruder) [inline src/test/java/uk/gov/hmcts/sptribs/caseworker/event/page/IssueFinalDecisionSelectRecipientsTest.java:68]: Could add an assert for expected error message returned.

File: src/test/java/uk/gov/hmcts/sptribs/caseworker/event/page/PostponeHearingNotifyPartiesTest.java [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+        final CaseData caseData = CaseData.builder().build();
+        caseDetails.setData(caseData);
+
+        final AboutToStartOrSubmitResponse<CaseData, State> response = postponeHearingNotifyParties.midEvent(caseDetails, caseDetails);
+
+        assertNull(response.getData().getCicCase().getNotifyPartySubject());
+        assertNull(response.getData().getCicCase().getNotifyPartyRepresentative());
+        assertNull(response.getData().getCicCase().getNotifyPartyRespondent());
+        assertNull(response.getData().getCicCase().getNotifyPartyApplicant());
+        assertThat(response.getErrors()).hasSize(1);
```
Comment (gmmagruder) [inline src/test/java/uk/gov/hmcts/sptribs/caseworker/event/page/PostponeHearingNotifyPartiesTest.java:70]: Could add an assert for expected error message returned.

File: src/test/java/uk/gov/hmcts/sptribs/caseworker/event/page/RecordNotifyPartiesTest.java [hunk 10]
```diff
@@ -50,10 +52,27 @@ void shouldBeInvalidIfNoRecipientsSelected() {
...
 
-        //When
         final AboutToStartOrSubmitResponse<CaseData, State> response = notifyParties.midEvent(caseDetails, caseDetails);
 
-        //Then
+        assertNull(response.getData().getCicCase().getNotifyPartySubject());
+        assertNull(response.getData().getCicCase().getNotifyPartyRepresentative());
+        assertNull(response.getData().getCicCase().getNotifyPartyRespondent());
+        assertNull(response.getData().getCicCase().getNotifyPartyApplicant());
         assertThat(response.getErrors()).hasSize(1);
```
Comment (gmmagruder) [inline src/test/java/uk/gov/hmcts/sptribs/caseworker/event/page/RecordNotifyPartiesTest.java:61]: Could add an assert for expected error message returned.

File: src/test/java/uk/gov/hmcts/sptribs/caseworker/event/page/ReinstateCaseNotifyPartiesTest.java [hunk 11]
```diff
@@ -3,30 +3,72 @@
...
         caseDetails.setData(caseData);
 
-        AboutToStartOrSubmitResponse<CaseData, State> response = reinstateNotifyParties.midEvent(caseDetails, caseDetails);
+        final AboutToStartOrSubmitResponse<CaseData, State> response = reinstateNotifyParties.midEvent(caseDetails, caseDetails);
 
+        assertNull(response.getData().getCicCase().getNotifyPartySubject());
+        assertNull(response.getData().getCicCase().getNotifyPartyRepresentative());
+        assertNull(response.getData().getCicCase().getNotifyPartyRespondent());
+        assertNull(response.getData().getCicCase().getNotifyPartyApplicant());
         assertThat(response.getErrors()).hasSize(1);
```
Comment (gmmagruder) [inline src/test/java/uk/gov/hmcts/sptribs/caseworker/event/page/ReinstateCaseNotifyPartiesTest.java:57]: Could add an assert for expected error message returned.

File: src/test/java/uk/gov/hmcts/sptribs/caseworker/event/page/SendOrderNotifyPartiesTest.java [hunk 12]
```diff
@@ -19,14 +30,49 @@ class SendOrderNotifyPartiesTest {
...
         caseDetails.setData(caseData);
 
-        AboutToStartOrSubmitResponse<CaseData, State> response = notifyParties.midEvent(caseDetails, caseDetails);
+        final AboutToStartOrSubmitResponse<CaseData, State> response = notifyParties.midEvent(caseDetails, caseDetails);
 
+        assertNull(response.getData().getCicCase().getNotifyPartySubject());
+        assertNull(response.getData().getCicCase().getNotifyPartyRepresentative());
+        assertNull(response.getData().getCicCase().getNotifyPartyRespondent());
+        assertNull(response.getData().getCicCase().getNotifyPartyApplicant());
         assertThat(response.getErrors()).hasSize(1);
```
Comment (gmmagruder) [inline src/test/java/uk/gov/hmcts/sptribs/caseworker/event/page/SendOrderNotifyPartiesTest.java:44]: Could add an assert for expected error message returned.

**Review by tomxelliott — DISMISSED** [review]

**Review by gmmagruder — DISMISSED** [review]

**Review by gmmagruder — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/sptribs/document/DocumentUtil.java [hunk 13]
```diff
@@ -92,17 +94,17 @@ public static void uploadDocument(CaseData data) {
     public static List<String> validateUploadedDocuments(List<ListValue<CaseworkerCICDocument>> uploadedDocuments) {
         List<String> errors = new ArrayList<>();
 
-        if (uploadedDocuments != null && !uploadedDocuments.isEmpty()) {
+        if (!CollectionUtils.isEmpty(uploadedDocuments)) {
```
Comment (rajaram-nagarajan) [inline src/main/java/uk/gov/hmcts/sptribs/document/DocumentUtil.java:97]: CollectionUtils empty check can be used in all other methods in this file as well.

File: src/test/java/uk/gov/hmcts/sptribs/caseworker/event/page/ContactPartiesSelectDocumentTest.java [hunk 14]
```diff
@@ -0,0 +1,67 @@
...
+    private ContactPartiesDocuments contactPartiesDocuments;
+
+    @Mock
+    private DynamicMultiSelectList dynamicMultiSelectList;
+
+    @Mock
+    private List<DynamicListElement> dynamicList;
+
+    @Test
+    void midEventIsSuccessful() {
```
Comment (rajaram-nagarajan) [inline src/test/java/uk/gov/hmcts/sptribs/caseworker/event/page/ContactPartiesSelectDocumentTest.java:38]: Add a test case for 'when no documents selected from the list'.

File: src/test/java/uk/gov/hmcts/sptribs/caseworker/event/page/IssueCaseNotifyPartiesTest.java [hunk 15]
```diff
@@ -0,0 +1,96 @@
...
+import static uk.gov.hmcts.sptribs.caseworker.util.ErrorConstants.SELECT_AT_LEAST_ONE_ERROR_MESSAGE;
+
+@ExtendWith(MockitoExtension.class)
+class IssueCaseNotifyPartiesTest {
+
+    @InjectMocks
+    private IssueCaseNotifyParties issueCaseNotifyParties;
+
+    @Test
+    void testMidEventReturnsErrorForNoRecipients() {
```
Comment (rajaram-nagarajan) [inline src/test/java/uk/gov/hmcts/sptribs/caseworker/event/page/IssueCaseNotifyPartiesTest.java:33]: test word can be removed from the method name.

File: src/test/java/uk/gov/hmcts/sptribs/caseworker/event/page/IssueCaseNotifyPartiesTest.java [hunk 16]
```diff
@@ -0,0 +1,96 @@
...
+            .cicCase(cicCase)
+            .build();
+        caseDetails.setData(caseData);
+
+        final AboutToStartOrSubmitResponse<CaseData, State> response = issueCaseNotifyParties.midEvent(caseDetails, caseDetails);
+        assertTrue(response.getErrors().isEmpty());
+    }
+
+    @Test
+    void midEventIsInvalidIfNoRecipientsSelected() {
```
Comment (rajaram-nagarajan) [inline src/test/java/uk/gov/hmcts/sptribs/caseworker/event/page/IssueCaseNotifyPartiesTest.java:60]: A new test case can be added to validate MINOR_FATAL_SUBJECT_ERROR_MESSAGE

File: src/test/java/uk/gov/hmcts/sptribs/caseworker/event/page/IssueCaseSelectDocumentTest.java [hunk 17]
```diff
@@ -0,0 +1,88 @@
...
+            .build();
+        caseDetails.setData(caseData);
+
+        AboutToStartOrSubmitResponse<CaseData, State> response = issueCaseSelectDocument.midEvent(caseDetails, caseDetails);
+
+        assertFalse(response.getErrors().isEmpty());
+        assertThat(response.getErrors()).contains("Select at least one document");
+    }
+
+    @Test
```
Comment (rajaram-nagarajan) [inline src/test/java/uk/gov/hmcts/sptribs/caseworker/event/page/IssueCaseSelectDocumentTest.java:71]: Add two more cases when the DocumentList is null or empty. You would need to move the check for 'Select at least one document' before 'Select up to 5 documents' check in the original java file.

File: src/test/java/uk/gov/hmcts/sptribs/caseworker/event/page/ManageSelectOrdersTest.java [hunk 18]
```diff
@@ -19,43 +21,72 @@
...
+        final CaseData caseData = CaseData.builder().build();
+        final Order order = Order.builder()
+            .dueDateList(List.of(mockListValueDateModel))
+            .draftOrder(new DraftOrderCIC())
+            .build();
+        final ListValue<Order> listValue = new ListValue<>();
+        listValue.setValue(order);
+        listValue.setId("0");
+        final CicCase cicCase = CicCase.builder()
+            .orderList(List.of(listValue))
```
Comment (rajaram-nagarajan) [inline src/test/java/uk/gov/hmcts/sptribs/caseworker/event/page/ManageSelectOrdersTest.java:75]: We may need to add new test cases when the orderlist is empty or null.

**Review by rajaram-nagarajan — CHANGES_REQUESTED** [review]

File: src/test/java/uk/gov/hmcts/sptribs/caseworker/event/page/ManageSelectOrdersTest.java (hunk 18 above)
Comment (oleon23) [reply src/test/java/uk/gov/hmcts/sptribs/caseworker/event/page/ManageSelectOrdersTest.java:78]: Added new test cases but had to amend ManageSelectOrders for NPE.

File: src/test/java/uk/gov/hmcts/sptribs/caseworker/event/page/IssueCaseSelectDocumentTest.java (hunk 17 above)
Comment (oleon23) [reply src/test/java/uk/gov/hmcts/sptribs/caseworker/event/page/IssueCaseSelectDocumentTest.java:87]: Added the new test cases and amended the midEvent as mentioned.

**Review by gmmagruder — DISMISSED** [review]

File: src/test/java/uk/gov/hmcts/sptribs/caseworker/event/page/IssueCaseSelectDocumentTest.java [hunk 19]
```diff
@@ -0,0 +1,103 @@
...
+        caseDetails.setData(caseData);
+
+        final AboutToStartOrSubmitResponse<CaseData, State> response = issueCaseSelectDocument.midEvent(caseDetails, caseDetails);
+
+        assertFalse(response.getErrors().isEmpty());
+        assertThat(response.getErrors()).contains("Select up to 5 documents");
+    }
+
+    @Test
+    void midEventReturnsNoErrorsForNullDocumentList() {
```
Comment (gmmagruder) [inline src/test/java/uk/gov/hmcts/sptribs/caseworker/event/page/IssueCaseSelectDocumentTest.java:73]: Change test name, test does expect an error

File: src/main/java/uk/gov/hmcts/sptribs/document/DocumentUtil.java (hunk 13 above)
Comment (gmmagruder) [inline src/main/java/uk/gov/hmcts/sptribs/document/DocumentUtil.java:97]: Can we use CollectionUtils.isNotEmpty instead for better readability and consistency

Comment (rajaram-nagarajan) [reply src/main/java/uk/gov/hmcts/sptribs/document/DocumentUtil.java:97]: Don't think isNotEmpty is available. Are you referring to use some other package?

Comment (oleon23) [reply src/main/java/uk/gov/hmcts/sptribs/document/DocumentUtil.java]: Thanks - updated.

**Review by gmmagruder — DISMISSED** [review]

File: src/test/java/uk/gov/hmcts/sptribs/citizen/event/CicCreateCaseEventTest.java [hunk 20]
```diff
@@ -55,20 +55,18 @@ public void setUp() {
...
+
         details.setId(TEST_CASE_ID);
-        CaseData data = caseData();
-        details.setData(data);
+        final CaseData caseData = caseData();
+        details.setData(caseData);
 
-        // When
-        AboutToStartOrSubmitResponse<CaseData, State> response = cicCreateCaseEvent.aboutToSubmit(
+        final AboutToStartOrSubmitResponse<CaseData, State> response = cicCreateCaseEvent.aboutToSubmit(
```
Comment (rajaram-nagarajan) [inline src/test/java/uk/gov/hmcts/sptribs/citizen/event/CicCreateCaseEventTest.java:65]: IsRepresentativePresent can be asserted for No.

File: src/test/java/uk/gov/hmcts/sptribs/citizen/event/CicCreateCaseEventTest.java [hunk 21]
```diff
@@ -55,20 +55,18 @@ public void setUp() {
...
             details,
             beforeDetails
-        );
+            );
 
-        // Then
```
Comment (rajaram-nagarajan) [inline src/test/java/uk/gov/hmcts/sptribs/citizen/event/CicCreateCaseEventTest.java:71]: A new test can be added with RepresentativeFullName and IsRepresentativePresent can be asserted for Yes.

File: src/test/java/uk/gov/hmcts/sptribs/citizen/event/CicSubmitCaseEventTest.java [hunk 22]
```diff
@@ -83,20 +84,26 @@ class CicSubmitCaseEventTest {
     @Mock
```
Comment (rajaram-nagarajan) [inline src/test/java/uk/gov/hmcts/sptribs/citizen/event/CicSubmitCaseEventTest.java:84]: public methods can be moved to the top in CicSubmitCaseEvent class

File: src/test/java/uk/gov/hmcts/sptribs/citizen/event/CicSubmitCaseEventTest.java [hunk 23]
```diff
@@ -187,16 +185,15 @@ void shouldUpdateCaseDetails() {
...
-        assertThat(response1.getData().getDssCaseData().getOtherInfoDocuments()).isEmpty();
-        assertThat(response1.getData().getCicCase().getApplicantDocumentsUploaded()).hasSize(2);
-        assertThat(response1.getData().getCicCase().getRepresentativeContactDetailsPreference()).isEqualTo(ContactPreferenceType.EMAIL);
-        assertThat(response1.getData().getCicCase().getApplicantDocumentsUploaded().get(0).getValue().getDocumentCategory())
+        assertThat(response).isNotNull();
+        assertThat(response.getData().getDssCaseData().getOtherInfoDocuments()).isEmpty();
+        assertThat(response.getData().getCicCase().getApplicantDocumentsUploaded()).hasSize(2);
+        assertThat(response.getData().getCicCase().getRepresentativeContactDetailsPreference()).isEqualTo(ContactPreferenceType.EMAIL);
+        assertThat(response.getData().getCicCase().getApplicantDocumentsUploaded().get(0).getValue().getDocumentCategory())
             .isIn(DocumentType.DSS_OTHER, DocumentType.DSS_SUPPORTING, DocumentType.DSS_TRIBUNAL_FORM);
```
Comment (rajaram-nagarajan) [inline src/test/java/uk/gov/hmcts/sptribs/citizen/event/CicSubmitCaseEventTest.java:197]: One test case can cover empty/null values for RepresentativeFullName, SupportingDocuments, TribunalFormDocuments and OtherInfoDocuments.
And another test case can cover with values for all.

File: src/test/java/uk/gov/hmcts/sptribs/citizen/event/CicUpdateCaseEventTest.java [hunk 24]
```diff
@@ -92,13 +93,12 @@ void shouldAddConfigurationToConfigBuilder() {
     }
 
     @Test
-    void shouldUpdateCaseDetails() {
```
Comment (rajaram-nagarajan) [inline src/test/java/uk/gov/hmcts/sptribs/citizen/event/CicUpdateCaseEventTest.java:95]: Arrays.asList(cicAppDetail) can be changed to List.of(cicAppDetail)

**Review by rajaram-nagarajan — CHANGES_REQUESTED** [review]

File: src/test/java/uk/gov/hmcts/sptribs/ciccase/model/CaseDataTest.java [hunk 25]
```diff
@@ -16,70 +16,61 @@ class CaseDataTest {
...
-        final CaseData caseData = CaseData.builder()
-            .build();
-        //Then
+        final CaseData caseData = CaseData.builder().build();
+
         assertThat(caseData.getCicCase().getApplicantFullName()).isNull();
     }
 
     @Test
     void shouldGetHearingDate() {
```
[the remark this answers is not in the document]
Comment (oleon23) [reply src/test/java/uk/gov/hmcts/sptribs/ciccase/model/CaseDataTest.java:25]: Created https://tools.hmcts.net/jira/browse/DTSSTCI-366 to resolve this.

File: src/test/java/uk/gov/hmcts/sptribs/ciccase/model/CaseDataTest.java [hunk 26]
```diff
@@ -16,70 +16,61 @@ class CaseDataTest {
...
-        //Then
         assertThat(result).isEqualTo(dateFormatter.format(now));
     }
 
     @Test
     void shouldGetHearingLocation() {
-        //When
-        String someAddress = "WC1";
-        Listing listing = Listing.builder()
+        final String someAddress = "WC1";
```
[the remark this answers is not in the document]
Comment (oleon23) [reply src/test/java/uk/gov/hmcts/sptribs/ciccase/model/CaseDataTest.java:42]: Created https://tools.hmcts.net/jira/browse/DTSSTCI-366 to resolve this.

File: src/test/java/uk/gov/hmcts/sptribs/ciccase/model/CaseDataTest.java [hunk 27]
```diff
@@ -16,70 +16,61 @@ class CaseDataTest {
...
+        final Listing listing = Listing.builder().hearingStatus(HearingState.Complete).date(now).build();
+        final ListValue<Listing> listingListValue = new ListValue<>();
         listingListValue.setValue(listing);
         final CaseData caseData = CaseData.builder()
             .hearingList(List.of(listingListValue))
             .build();
 
         //When
-        Listing result = caseData.getLatestCompletedHearing();
+        final Listing result = caseData.getLatestCompletedHearing();
```
[the remark this answers is not in the document]
Comment (oleon23) [reply src/test/java/uk/gov/hmcts/sptribs/ciccase/model/CaseDataTest.java:70]: Created https://tools.hmcts.net/jira/browse/DTSSTCI-366 to resolve this.

File: src/test/java/uk/gov/hmcts/sptribs/ciccase/model/CaseDataTest.java [hunk 28]
```diff
@@ -16,70 +16,61 @@ class CaseDataTest {
...
 
         //Then
         assertThat(result).isNotNull();
+        assertThat(result.getDate()).isEqualTo(now);
```
[the remark this answers is not in the document]
Comment (oleon23) [reply src/test/java/uk/gov/hmcts/sptribs/ciccase/model/CaseDataTest.java:74]: Created https://tools.hmcts.net/jira/browse/DTSSTCI-366 to resolve this.

File: src/test/java/uk/gov/hmcts/sptribs/citizen/event/CicCreateCaseEventTest.java (hunk 20 above)
Comment (oleon23) [reply src/test/java/uk/gov/hmcts/sptribs/citizen/event/CicCreateCaseEventTest.java:65]: Created https://tools.hmcts.net/jira/browse/DTSSTCI-366 to resolve this.

File: src/test/java/uk/gov/hmcts/sptribs/citizen/event/CicCreateCaseEventTest.java (hunk 21 above)
Comment (oleon23) [reply src/test/java/uk/gov/hmcts/sptribs/citizen/event/CicCreateCaseEventTest.java:71]: Created https://tools.hmcts.net/jira/browse/DTSSTCI-366 to resolve this.

File: src/test/java/uk/gov/hmcts/sptribs/citizen/event/CicSubmitCaseEventTest.java (hunk 22 above)
Comment (oleon23) [reply src/test/java/uk/gov/hmcts/sptribs/citizen/event/CicSubmitCaseEventTest.java:84]: Created https://tools.hmcts.net/jira/browse/DTSSTCI-366 to resolve this.

File: src/test/java/uk/gov/hmcts/sptribs/citizen/event/CicSubmitCaseEventTest.java (hunk 23 above)
Comment (oleon23) [reply src/test/java/uk/gov/hmcts/sptribs/citizen/event/CicSubmitCaseEventTest.java:197]: Created https://tools.hmcts.net/jira/browse/DTSSTCI-366 to resolve this.

File: src/test/java/uk/gov/hmcts/sptribs/citizen/event/CicUpdateCaseEventTest.java (hunk 24 above)
Comment (oleon23) [reply src/test/java/uk/gov/hmcts/sptribs/citizen/event/CicUpdateCaseEventTest.java:95]: Created https://tools.hmcts.net/jira/browse/DTSSTCI-366 to resolve this.

**Review by tomxelliott — APPROVED** [review]
Looks good to me. Happy for the comments provided by @rajaram-nagarajan to be tackled in new ticket

**Review by rajaram-nagarajan — APPROVED** [review]
LGTM
